satellite/metainfo/service.go

```diff
@@ -43,28 +42,110 @@ func (s *Service) Put(ctx context.Context, path string, pointer *pb.Pointer) (er
 		return Error.Wrap(err)
 	}
 
-	// TODO(kaloyan): make sure that we know we are overwriting the pointer!
-	// In such case we should delete the pieces of the old segment if it was
-	// a remote one.
-	if err = s.DB.Put(ctx, []byte(path), pointerBytes); err != nil {
-		return Error.Wrap(err)
-	}
+	// CompareAndSwap is used instead of Put to avoid overwriting existing pointers
+	err = s.DB.CompareAndSwap(ctx, []byte(path), nil, pointerBytes)
```

Comment:
With this change this `Put` method behaves like an `Add` isn't it?

Reply:
It acts like the `Put` before, but it will fail with `storage.ErrValueChanged` if such a path already exists. This way we avoid overriding of pointers, which can lead to serious issues.

Reply:
Then it's always inserted, so it's like an `Add`.
I don't see `Put` as a good name after this has changed, but obviously renaming the method will break the public API of the package requiring to update many other packages.